### PR TITLE
Test "device" and "pre_DataConsent" must be in PT

### PR DESCRIPTION
**Pull request by orhan-mehmedov**
Enabled policy_manager_impl_stress_test.cc and
policy_manager_impl_test.cc.
Test updated to test device and pre_DataConsent
presence in PT. Also some other tests improved.

Related to APPLINK-26927.

### CODE REVIEW

**Comment by orhan-mehmedov** [comment]
Please review,
@AByzhynar @LuxoftAKutsan @anosach-luxoft @dev-gh @Kozoriz @pvvasilev
Optional: @pestOO 
//wolfylambova22 is on vacation

File: src/components/policy/test/policy_manager_impl_stress_test.cc [hunk 1]
```diff
@@ -47,8 +47,8 @@ namespace policy {
 class PolicyManagerImplStressTest : public ::testing::Test {
  protected:
   static const std::string kNameFile;
-  static const int kNumberGroups = 3;  //10;
-  static const int kNumberFuncs = 4;  //100;
+  static const int kNumberGroups = 3;  // 10;
```
Comment (LuxoftAKutsan) [inline src/components/policy/test/policy_manager_impl_stress_test.cc]: @orhan-mehmedov This comments are confused

File: src/components/policy/test/policy_manager_impl_test.cc [hunk 2]
```diff
@@ -65,11 +68,11 @@ namespace components {
 namespace policy {
 
 class PolicyManagerImplTest : public ::testing::Test {
-protected:
-  PolicyManagerImpl *manager;
-  MockCacheManagerInterface *cache_manager;
+ protected:
+  PolicyManagerImpl* manager;
```
Comment (LuxoftAKutsan) [inline src/components/policy/test/policy_manager_impl_test.cc]: @orhan-mehmedov is it possible to store this values by value or reference ot auto pointer or even shared pointer?

File: src/components/policy/test/policy_manager_impl_test.cc [hunk 3]
```diff
@@ -86,7 +89,44 @@ class PolicyManagerImplTest : public ::testing::Test {
...
+      return ::testing::AssertionSuccess();
+    } else {
+      ::rpc::ValidationReport report(" - table");
+      table.ReportErrors(&report);
+      return ::testing::AssertionFailure() << ::rpc::PrettyFormat(report);
+    }
+  }
+};
+
+class PolicyManagerImplTest2 : public ::testing::Test {
```
Comment (LuxoftAKutsan) [inline src/components/policy/test/policy_manager_impl_test.cc]: @orhan-mehmedov Please at least add comments why do you need PolicyManagerImplTest2?

File: src/components/policy/test/policy_manager_impl_test.cc [hunk 4]
```diff
@@ -277,57 +338,66 @@ TEST_F(PolicyManagerImplTest, LoadPT_SetPT_PTIsLoaded) {
...
+  std::ifstream ifile("sdl_preloaded_pt.json");
+  Json::Reader reader;
+  Json::Value root(Json::objectValue);
 
-    // arrange
-    ::utils::SharedPtr<::policy_table::Table> p_table =
-        new ::policy_table::Table();
+  if (ifile.is_open() && reader.parse(ifile, root, true)) {
+    Json::Value seconds_between_retries = Json::Value(Json::arrayValue);
+    seconds_between_retries =
```
Comment (LuxoftAKutsan) [inline src/components/policy/test/policy_manager_impl_test.cc]: @orhan-mehmedov is it possible = root["policy_table"]["module_config"]["seconds_between_retries"];

File: src/components/policy/test/policy_manager_impl_test.cc [hunk 5]
```diff
@@ -86,7 +89,44 @@ class PolicyManagerImplTest : public ::testing::Test {
...
+
+class PolicyManagerImplTest2 : public ::testing::Test {
+ protected:
+  PolicyManagerImpl* manager;
+  MockCacheManagerInterface* cache_manager;
+  MockUpdateStatusManager update_manager;
+  MockPolicyListener* listener;
+
+  void CreateLocalPT(const std::string& filename) {
+    // what to do here ?
```
Comment (LuxoftAKutsan) [inline src/components/policy/test/policy_manager_impl_test.cc]: @orhan-mehmedov Please remove this comment. Please TODO or implement  this method.

File: src/components/policy/test/policy_manager_impl_test.cc [hunk 6]
```diff
@@ -277,57 +338,66 @@ TEST_F(PolicyManagerImplTest, LoadPT_SetPT_PTIsLoaded) {
...
-        new ::policy_table::Table();
+  if (ifile.is_open() && reader.parse(ifile, root, true)) {
+    Json::Value seconds_between_retries = Json::Value(Json::arrayValue);
+    seconds_between_retries =
+        root["policy_table"]["module_config"]["seconds_between_retries"];
+    uint32_t size = seconds_between_retries.size();
 
-    // assert
-    EXPECT_CALL(*cache_manager, GenerateSnapshot()).WillOnce(Return(p_table));
+    CreateLocalPT("sdl_preloaded_pt.json");
```
Comment (LuxoftAKutsan) [inline src/components/policy/test/policy_manager_impl_test.cc]: @orhan-mehmedov as I remember this method is empty, are you sure that it is to call it?

File: src/components/policy/test/policy_manager_impl_test.cc [hunk 7]
```diff
@@ -277,57 +338,66 @@ TEST_F(PolicyManagerImplTest, LoadPT_SetPT_PTIsLoaded) {
...
+      waiting_timeout += manager->TimeoutExchange();
 
-  TEST_F(PolicyManagerImplTest, DISABLED_GetPolicyTableStatus) {
-    // TODO(AOleynik): Test is not finished, to be continued
-    // manager->GetPolicyTableStatus();
+      // it's in milliseconds
+      EXPECT_EQ(waiting_timeout * date_time::DateTime::MILLISECONDS_IN_SECOND,
+                manager->NextRetryTimeout());
+    }
+    ifile.close();
```
Comment (LuxoftAKutsan) [inline src/components/policy/test/policy_manager_impl_test.cc]: @orhan-mehmedov wan't it be closed in destructor?

File: src/components/policy/test/policy_manager_impl_test.cc [hunk 8]
```diff
@@ -277,57 +338,66 @@ TEST_F(PolicyManagerImplTest, LoadPT_SetPT_PTIsLoaded) {
...
+  // assert
+  EXPECT_CALL(*cache_manager, GenerateSnapshot()).WillOnce(Return(p_table));
+  EXPECT_CALL(*listener, OnSnapshotCreated(_));
+
+  // act
+  manager->RequestPTUpdate();
+}
+
+TEST_F(PolicyManagerImplTest, DISABLED_AddApplication) {
+  // TODO(AOleynik): Implementation of method should be changed to avoid
```
Comment (LuxoftAKutsan) [inline src/components/policy/test/policy_manager_impl_test.cc]: @orhan-mehmedov please add tasks for TODOs

**Comment by orhan-mehmedov** [comment]
Changes of UT related to APPLINK-26927 will be done in development branch.
No UT updates will be delivered to master branch so this PR is not needed and will be closed.

**Pull request closed without merging**